### PR TITLE
Feature/wallet connect

### LINKED ISSUE #156: Landing Page | EVM Wallet - connect to EVM Wallet
## Overview
 As a user I want to connect my EVM account using metamask and walletconnect
## Acceptance Criteria
- [x] display Connect Wallet button
- [x] display modal with button to select metamask and  button to select walletconnect on desktop, walletconnect list should not show metamask<sup>1</sup>, it also includes x button that gets user to landing page view
- [x] on mobile directly link connect wallet button to walletconnect modal display and flow
- [x] on desktop if metamask is not installed, ~~show modal that provides button~~<sup>2</sup> redirect to install (link to [metamask install flow](https://metamask.io/download/)
  ~~- [ ]  include a X button that gets user back to landing page and~~
  ~~- [ ]  a Choose another wallet button that will go back to choose wallet modal.~~
- [x] Login should redirect to balance page
- [x] Logout should disconnect wallet connect (after logout, connecting via WalletConnect again should _not_ show 'connected' modal)

Notes:
<sup>1</sup> if wallets are installed and enabled (metamask) they are shown by default (not excluded)
<sup>2</sup> in lieu of an additional modal, I changed the button text to 'download metamask' and redirect on click

### PR DESCRIPTION
**Pull request by donnyquixotic**
# Fixes #156 

## Description
See issue description

## Test scenarios
Mobile:

1.   On click 'Connect Your Wallet' it opens WalletConnect modal
2.   Selecting mobile wallet from available list should prompt sign-in, chain-switching (if nec) and redirect to 'balances' page
3.   Refreshing page without signing out should go directly to balances page
4.   Log out, refresh, should display login page

Desktop:

1.  Clicking 'Connect Your Wallet' button should open a new dialog with 'MetaMask' and 'WalletConnect' and button options
2.  Clicking the 'x' should close the dialog and return to the original homescreen
5.  If all wallet extensions are disabled, MetaMask button should read 'Download MetaMask' 
6.  Clicking should redirect to download page in a new tab
7.  If MetaMask is installed, MetaMask button should read 'MetaMask' and prompt connection in the browser extension
8.  Clicking WalletConnect button should open WalletConnect modal
9.  If Metamask is not installed or enabled, it should be ommitted from the available wallet selection in the WalletConnect list
10. Selecting an option should prompt connection, and redirect to 'balances' page
11. Refreshing page without signing out should go directly to balances page
12. Log out, refresh, should display login page

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

File: env.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const sharedEnv = {
     GOOGLE_ANALYTICS: 'UA-154600181-2',
     IMGUR_CLIENT_ID: 'b6f46df9d1da9d9',
     EVM_CONTRACT: 'eosio.evm',
+    PROJECT_ID: '14ec76c44bae7d461fa0f5fd5f8a9da1',
```
Comment (EJSG) [inline env.js:11]: just curious, where does this come from?

**Review by EJSG — APPROVED** [review]
works well for me, really awesome work. tested on android and desktop. the wallet connect UX is impressive

**Comment by Viterbo** [comment]
This happened to me. It says Install Metamask but I already got it. I'm going to investigate a little more what is happening. I'm running a local copy using your branch. Gonna take a look and come back with more feedback
image

File: src/pages/home/ConnectWalletOptions.vue [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            Connect your wallet
+        </div>
+        <div class="wallet-options__option" @click="supportsMetamask ? loginEvm() : redirectToMetamaskDownload()">
+            <img
+                width="24"
+                class="flex q-ml-auto q-mt-auto wallet-logo"
+                alt="MetaMask"
+                src="~assets/evm/metamask_fox.svg"
+            >
+            {{ supportsMetamask ? 'MetaMask' : 'Install MetaMask' }}
```
Comment (Viterbo) [inline src/pages/home/ConnectWalletOptions.vue:101]:
```suggestion
            {{ supportsMetamask ? $t('home.metamask') : $t('home.install_metamask') }}
```

File: src/pages/home/ConnectWalletOptions.vue [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        class="wallet-options__close"
+        icon="close"
+        flat
+        round
+        dense
+        @click="$emit('closeWalletOptions')"
+    />
+    <div class="wallet-options">
+        <div class="wallet-options__header">
+            Connect your wallet
```
Comment (Viterbo) [inline src/pages/home/ConnectWalletOptions.vue:92]:
```suggestion
            {{ $t('home.connect_your_wallet') }}
```

File: src/pages/home/ConnectWalletOptions.vue [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+            {{ supportsMetamask ? 'MetaMask' : 'Install MetaMask' }}
+        </div>
+        <div class="wallet-options__option" @click="connectToWalletConnect">
+            <img
+                width="24"
+                class="flex q-ml-auto q-mt-auto wallet-logo"
+                alt="WalletConnect"
+                src="~assets/evm/wallet_connect.svg"
+            >
+            WalletConnect
```
Comment (Viterbo) [inline src/pages/home/ConnectWalletOptions.vue:110]:
```suggestion
            {{ $t('home.walletconnect') }}
```

**Review by Viterbo — CHANGES_REQUESTED** [review]
Add the following new properties into the `src/i18n/en-us/index.js` file under the `home` section
```js
    install_metamask: 'Install MetaMask',
    metamask: 'MetaMask',
    walletconnect: 'WalletConnect',
    connect_your_wallet: 'Connect Your Wallet',
```

**Comment by pmjanus** [comment]
> Member
I had this happen and was not able to reproduce.

File: src/pages/home/ConnectWalletOptions.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    props: {
+        toggleWalletConnect: {
+            required: true,
+            type: Boolean,
+        },
+    },
+    setup(props){
+        const wagmiClient = inject('$wagmi') as EthereumClient;
+        const web3Modal = ref<Web3Modal>();
+        const supportsMetamask = computed(() => useEVMStore().isMetamaskSupported);
```
Comment (Viterbo) [inline src/pages/home/ConnectWalletOptions.vue:23]: This path solves the problem [found here](https://github.com/telosnetwork/telos-wallet/pull/239#issuecomment-1509438649)

```suggestion
        const supportsMetamask = computed(() =>
            useEVMStore().isMetamaskSupported || localStorage.getItem('wagmi.connected'));
```

File: src/antelope/stores/evm.ts [hunk 6]
```diff
@@ -80,6 +81,11 @@ export const useEVMStore = defineStore(store_name, {
         trace: createTraceFunction(store_name),
         init: () => {
 
+            // bypass provider detection if using WalletConnect
+            if (localStorage.getItem('wagmi.connected')){
+                return;
+            }
```
Comment (Viterbo) [inline src/antelope/stores/evm.ts:87]: this line prevents` evm.setSupportsMetaMask(true);` from happening, and the home page is asking const `supportsMetamask = computed(() => useEVMStore().isMetamaskSupported` which returns false and that generates the button to show the "Install Metamask" label

**Review by Viterbo — CHANGES_REQUESTED** [review]
The path solves the problem but I will create an issue to refactor things to have it all integrated into the new store.

File: src/pages/home/ConnectWalletOptions.vue (hunk 5 above)
Comment (donnyquixotic) [reply src/pages/home/ConnectWalletOptions.vue:22]: resolved

File: src/antelope/stores/evm.ts (hunk 6 above)
Comment (donnyquixotic) [reply src/antelope/stores/evm.ts]: Changed this to only return on mobile which resolves the issue.

**Pull request merged**